src/memory_dump/syscall_interface.c

```diff
@@ -0,0 +1,241 @@
+#include <assert.h>
+#include <sys/syscall.h>
+#include <stdbool.h>
+#include <stdarg.h>
+#include <fcntl.h>
+#include <sys/mman.h>
+
+#include "types.h"
+#include "pages.h"
+#include "ptrace.h"
+#include "syscall_interface.h"
+
+#define _DEBUG 1
+#undef _DEBUG
+
+#include "debug.h"
+
+#ifdef __x86_64__
+static register_t inject_syscall(int syscallid, pid_t tid, struct user_regs_struct *regs, va_list vargs) {
```

Comment:
`inject_syscall`, `injection_code` and `interface.h` can be simplified.

You  have separated cases in `interface.h` which all call the generic `injection_code`, then you need to switch/case in `inject_syscall`. Can we remove one of the levels of abstraction, and remove the case switch here ?